cli/crud/lua/import.lua

```diff
@@ -0,0 +1,465 @@
+-- This script implements logic on the router side for working with crud as part of the crud-import task.
+-- For communication between the code on golang and lua, evals are initialized in the session storage,
+-- which will be called from golang via tt connect.
+
+
+local box  = require('box')
+local crud = require('crud')
+-- NOTE: crud issue #293, we don't need ddl module now, but in the future it may be.
+-- local ddl  = require('ddl')
+local json = require('json')
+local decimal = require('decimal')
+local ffi = require('ffi')
+
+local crud_evals_init_complete = false
+box.session.storage.batch_insert_res = nil
+box.session.storage.batch_insert_err = nil
+box.session.storage.null_val_interpretation = nil
+
+-- Map string to boolean if possible, else return nil.
+local toboolean = function(value)
+    local map = {
+        ['true']  = true,
+        ['false'] = false,
+        ['t']  = true,
+        ['f']  = false
+    }
+    if map[value] ~= nil then
+        return map[value]
+    end
+
+    return nil
+end
+
+-- Set crud.insert_many as crud stored procedure for import.
+local init_eval_set_batch_insert_stored_procedure = function()
+    box.session.storage.crudimport_batch_stored_procedure = crud.insert_many
+end
+
+-- Set crud.replace_many as crud stored procedure for import.
+local init_eval_set_batch_replace_stored_procedure = function()
+    box.session.storage.crudimport_batch_stored_procedure = crud.replace_many
+end
+
+-- Set one of available crud stored procedure for import.
+local init_eval_set_stored_procedure = function()
+    box.session.storage.crudimport_set_stored_procedure = function(name)
+        if name == 'insert' then
+            init_eval_set_batch_insert_stored_procedure()
+            if type(box.session.storage.crudimport_batch_stored_procedure) ~= "function" then
+                return "Function crud.insert_many in not found."
+            end
+            return true
+        end
+        if name == 'replace' then
+            init_eval_set_batch_replace_stored_procedure()
+            if type(box.session.storage.crudimport_batch_stored_procedure) ~= "function" then
+                return "Function crud.replace_many in not found."
+            end
+            return true
+        end
+
+        return "Unknown crud stored procedure, check operation option."
+    end
+end
+
+-- Set target space for import.
+local init_eval_set_targetspace = function()
+    box.session.storage.crudimport_set_targetspace = function(space)
+        box.session.storage.crudimport_targetspace = space
+    end
+end
+
+-- Check existance of target space for import.
+local init_eval_check_targetspace_exist = function()
+    box.session.storage.crudimport_check_targetspace_exist = function()
+        if crud.select(box.session.storage.crudimport_targetspace) ~= nil then
+            box.session.storage.crudimport_targetspace_format =
+                crud.select(box.session.storage.crudimport_targetspace)['metadata']
```

Comment:
You already makes this request a line above. It is better to save the result and reuse it instead of sending the request twice.

Reply:
Done